burn-tensor/src/tensor/ops/modules/conv.rs

```diff
@@ -76,36 +59,30 @@ pub(crate) fn conv1d_backward<B: Backend>(
         output_grad.clone(),
         weight,
         None,
-        stride,
-        padding,
-        padding_out,
-        dilation,
-    );
-
-    let x_swapped = B::swap_dims(x, 0, 1);
-    let output_grad_swapped = B::swap_dims(output_grad, 0, 1);
-    let weight_grad_swapped = B::conv1d(
-        x_swapped,
-        output_grad_swapped.clone(),
-        None,
-        dilation,
-        padding,
-        stride,
+        ConvTransposeOptions::new(
+            options.stride,
+            options.padding,
+            [padding_out],
+            options.dilation,
+            options.groups,
+        ),
     );
-    let mut weight_grad = B::swap_dims(weight_grad_swapped, 0, 1);
 
-    if B::shape(&weight_grad) != Shape::new([channels_out, channels_in, kernel_size]) {
-        weight_grad = B::index(
-            weight_grad,
-            [0..channels_out, 0..channels_in, 0..kernel_size],
-        );
-    }
+    let weight_grad = match options.groups == 1 {
+        true => conv1d_weight_grad_no_groups::<B>(x, output_grad.clone(), weight_shape, options),
+        false => conv1d_weight_grad_groups::<B>(
+            x,
+            B::zeros(weight_shape, &weight_device),
+            output_grad.clone(),
+            options,
+        ),
+    };
```

Comment:
if else would have sufficed =D

Reply:
Yup, I just prefer the see the two functions called with no line between the two =D